spec/controllers/prompt_meme_controller_spec.rb

```diff
@@ -3,6 +3,46 @@
 describe Challenge::PromptMemeController do
   include LoginMacros
 
+  describe "new" do
+    context "when the collection already has a challenge" do
+      before do
+        @collection = FactoryGirl.create(:collection, challenge: PromptMeme.new)
+        fake_login_known_user(@collection.owners.first.user)
+        post :new, params: { collection_id: @collection.name }
+      end
+
+      it "should set flash notice" do
+        expect(flash[:notice]).to eq("There is already a challenge set up for this collection.")
+      end
+
+      it "should redirect to edit meme collection path" do
+        expect(response).to redirect_to(edit_collection_prompt_meme_path(@collection))
+      end
+    end
+  end
+
+  describe "update" do
+    context "when it fails to udpate parameters" do
+      before do
+        challenge = PromptMeme.new
```

Comment:
It looks like the only reference to the `challenge` variable here is in `@collection = FactoryGirl.create(:collection, challenge: challenge)`. Would it be possible to avoid creating the variable and do `FactoryGirl.create(:collection, challenge: PromptMeme.new)` here like in the test above? 

You can leave off `FactoryGirl` for brevity if you like and it should still work.